hazelcast/src/main/java/com/hazelcast/spi/impl/servicemanager/RemoteServiceDescriptorProvider.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (c) 2008-2016, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.spi.impl.servicemanager;
+
+/***
+ * Provider for the remote service descriptors;
+ * Will be read from META-INF/services
+ */
+public interface RemoteServiceDescriptorProvider {
```

Comment:
This class is public API AFAIK and should be moved somewhere outside of an impl package.

Can you add some documentation how this class fits into the big picture? If I'm a third party developer, I want to create my own e.g. Tree Service.. then I need to implement this class and do something with it (put it in special location).